### ISSUE TITLE
Impossibly high Mbps when running Netmap on X540 

### ISSUE THREAD

**Issue by fklassen** [appneta/tcpreplay#427]
As per issue reported in mailing list ....

```
On ubuntu 16 (Intel D-1500 SoC) using tcpreplay version 4.2.5 (build git:v4.2.5) testing thru an Intel X520 10Gbe, when using the "-t" option it's reporting throughput of > 10Gbs:

root@send:/home/mzealadmin# taskset -c 0 tcpreplay -t --intf1=enp6s0f1 -K --netmap --no-flow-stats large.pcap 
Switching network driver for enp6s0f1 to netmap bypass mode... done!
File Cache is enabled
Actual: 81706489 packets (42796983763 bytes) sent in 20.65 seconds
Rated: 2072292454.1 Bps, 16578.33 Mbps, 3956347.52 pps
Statistics for network device: enp6s0f1
    Successful packets:        81706489
    Failed packets:            0
    Truncated packets:         0
    Retried packets (ENOBUFS): 0
    Retried packets (EAGAIN):  15165240
Switching network driver for enp6s0f1 to normal mode... done!
```

**Comment by fklassen** [comment]
Comment from Aaron...

```
I'm looking at the output and noticing a lot of EAGAIN errors.  I
wonder if those are being counted towards the Mbps calc?
--
Aaron Turner
https://synfin.net/         Twitter: @synfinatic
```

**Comment by ken-adey** [comment]
I can't find any version info in the netmap distribution, but I downloaded it from https://github.com/luigirizzo/netmap a month ago, so it's probably the latest.

If I use the -M option instead, eg:

root@send:/home/mzealadmin# taskset -c 0 tcpreplay -M 9000 --intf1=enp6s0f1 -K --netmap --no-flow-stats large.pcap 
Switching network driver for enp6s0f1 to netmap bypass mode... done!
File Cache is enabled
Actual: 81706489 packets (42796983763 bytes) sent in 38.04 seconds
Rated: 1125022574.6 Bps, 9000.18 Mbps, 2147853.34 pps
Statistics for network device: enp6s0f1
	Successful packets:        81706489
	Failed packets:            0
	Truncated packets:         0
	Retried packets (ENOBUFS): 0
	Retried packets (EAGAIN):  1993
Switching network driver for enp6s0f1 to normal mode... done!

there are a lot fewer EAGAIN errors. So can I assume the fewer EAGAIN errors reported, the more accurate the "Rated" stats are?

**Comment by fklassen** [comment]
`EAGAIN` errors are not really errors, and do not affect performance if handled properly. It says that the TX buffers are full, and the program should retry sending.

**Comment by fklassen** [comment]
Reported [this issue in netmap repo](https://github.com/luigirizzo/netmap/issues/439). Deferring until I get further info.

**Comment by fklassen** [comment]
I suspect that fix for #424 will fix this issue.

It appears that newer versions of netmap were not reporting when a TX buffer was empty. Therefore Tcpreplay thinks that all packets have left the adapter, when in fact they are still buffered. The "end" timestamp happens early, and capacity is too high.

I'll close this issue. If this exact condition happens again, feel free to reopen.
